core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -162,7 +162,9 @@ HIPInternal &HIPInternal::singleton() {
   return *self;
 }
 
-void HIPInternal::fence() const { hipStreamSynchronize(m_stream); }
+void HIPInternal::fence() const {
+  HIP_SAFE_CALL(hipStreamSynchronize(m_stream));
```

Comment:
This would be called from https://github.com/kokkos/kokkos/blob/66e00acdf82c167cb5b3dfe80625e4ffbe6a4829/core/src/HIP/Kokkos_HIP_Space.cpp#L750 and solve the issue reported in #3243

Reply:
Pleas check the error code in Cuda as well